packages/xo-web/src/xo-app/backup/restore/index.js

```diff
@@ -184,9 +143,53 @@ export default class Restore extends Component {
       backupDataByVm[vmId].backups = orderBy(backups, 'timestamp', 'desc')
     })
 
-    this.setState({ backupDataByVm })
+    return backupDataByVm
   }
 
+  __refreshBackupList = async (remote, jobs) => {
+    const remoteId = remote.id
+    const backupsByRemote = await listVmBackups([remoteId])
+    const { backupDataByVm } = this.state
+    const remoteBackupDataByVm = {}
+    forEach(backupsByRemote[remoteId], (vmBackups, vmId) => {
+      if (vmBackups.length === 0) {
+        return
+      }
+
+      const backupData = backupDataByVm[vmId]
+      remoteBackupDataByVm[vmId] =
+        backupData === undefined ? { backups: [] } : backupData
+
+      remoteBackupDataByVm[vmId].backups.push(
+        ...map(vmBackups, bkp => {
+          const job = find(jobs, { id: bkp.jobId })
+          return {
+            ...bkp,
+            remote,
+            jobName: job && job.name,
+          }
+        })
+      )
+    })
+
+    this.setState(({ backupDataByVm }) => ({
+      backupDataByVm: {
+        ...backupDataByVm,
+        ...this._sortBackupList(remoteBackupDataByVm),
+      },
+    }))
+  }
+
+  _refreshBackupList = (
+    _remotes = this.props.remotes,
+    jobs = this.props.jobs
+  ) =>
+    Promise.all(
+      map(filter(_remotes, { enabled: true }), remote =>
+        this.__refreshBackupList(remote, jobs)
```

Comment:
What happens if one of the calls fail?

Reply:
After fixes, we display an error notification.